[browser][MT] unify pthread pool size to 32

pavelsavara opened the pull request:
- bump default thread pool size to 32 and unify to that default for all tests
 - see which existing active issues are caused by the same reason
 - keep ActiveIssue tracking it, because we know that it's too large default
     - https://github.com/dotnet/runtime/issues/91538

pavelsavara commented:
/azp run runtime-wasm

pavelsavara committed:
wip

pavelsavara commented:
/azp run runtime-wasm

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/arcade dotnet/xharness dotnet/sdk (#93929)

* Update dependencies from https://github.com/dotnet/arcade build 20231023.2

Microsoft.DotNet.Arcade.Sdk , Microsoft.DotNet.Build.Tasks.Archives , Microsoft.DotNet.Build.Tasks.Feed , Microsoft.DotNet.Build.Tasks.Installers , Microsoft.DotNet.Build.Tasks.Packaging , Microsoft.DotNet.Build.Tasks.TargetFramework , Microsoft.DotNet.Build.Tasks.Templating , Microsoft.DotNet.Build.Tasks.Workloads , Microsoft.DotNet.CodeAnalysis , Microsoft.DotNet.GenAPI , Microsoft.DotNet.GenFacades , Microsoft.DotNet.Helix.Sdk , Microsoft.DotNet.PackageTesting , Microsoft.DotNet.RemoteExecutor , Microsoft.DotNet.SharedFramework.Sdk , Microsoft.DotNet.VersionTools.Tasks , Microsoft.DotNet.XUnitAssert , Microsoft.DotNet.XUnitConsoleRunner , Microsoft.DotNet.XUnitExtensions
 From Version 9.0.0-beta.23518.2 -> To Version 9.0.0-beta.23523.2

* Update dependencies from https://github.com/dotnet/arcade build 20231024.1

Microsoft.DotNet.Arcade.Sdk , Microsoft.DotNet.Build.Tasks.Archives , Microsoft.DotNet.Build.Tasks.Feed , Microsoft.DotNet.Build.Tasks.Installers , Microsoft.DotNet.Build.Tasks.Packaging , Microsoft.DotNet.Build.Tasks.TargetFramework , Microsoft.DotNet.Build.Tasks.Templating , Microsoft.DotNet.Build.Tasks.Workloads , Microsoft.DotNet.CodeAnalysis , Microsoft.DotNet.GenAPI , Microsoft.DotNet.GenFacades , Microsoft.DotNet.Helix.Sdk , Microsoft.DotNet.PackageTesting , Microsoft.DotNet.RemoteExecutor , Microsoft.DotNet.SharedFramework.Sdk , Microsoft.DotNet.VersionTools.Tasks , Microsoft.DotNet.XUnitAssert , Microsoft.DotNet.XUnitConsoleRunner , Microsoft.DotNet.XUnitExtensions
 From Version 9.0.0-beta.23518.2 -> To Version 9.0.0-beta.23524.1

Dependency coherency updates

Microsoft.DotNet.XliffTasks
 From Version 1.0.0-beta.23516.1 -> To Version 1.0.0-beta.23523.1 (parent: Microsoft.DotNet.Helix.Sdk

* Update dependencies from https://github.com/dotnet/arcade build 20231024.1

Microsoft.DotNet.Arcade.Sdk , Microsoft.DotNet.Build.Tasks.Archives , Microsoft.DotNet.Build.Tasks.Feed , Microsoft.DotNet.Build.Tasks.Installers , Microsoft.DotNet.Build.Tasks.Packaging , Microsoft.DotNet.Build.Tasks.TargetFramework , Microsoft.DotNet.Build.Tasks.Templating , Microsoft.DotNet.Build.Tasks.Workloads , Microsoft.DotNet.CodeAnalysis , Microsoft.DotNet.GenAPI , Microsoft.DotNet.GenFacades , Microsoft.DotNet.Helix.Sdk , Microsoft.DotNet.PackageTesting , Microsoft.DotNet.RemoteExecutor , Microsoft.DotNet.SharedFramework.Sdk , Microsoft.DotNet.VersionTools.Tasks , Microsoft.DotNet.XUnitAssert , Microsoft.DotNet.XUnitConsoleRunner , Microsoft.DotNet.XUnitExtensions
 From Version 9.0.0-beta.23518.2 -> To Version 9.0.0-beta.23524.1

Dependency coherency updates

Microsoft.DotNet.XliffTasks
 From Version 1.0.0-beta.23516.1 -> To Version 1.0.0-beta.23523.1 (parent: Microsoft.DotNet.Helix.Sdk

* Update dependencies from https://github.com/dotnet/xharness build 20231025.2

Microsoft.DotNet.XHarness.CLI , Microsoft.DotNet.XHarness.TestRunners.Common , Microsoft.DotNet.XHarness.TestRunners.Xunit
 From Version 8.0.0-prerelease.23523.1 -> To Version 8.0.0-prerelease.23525.2

* Update dependencies from https://github.com/dotnet/sdk build 20231025.7

Microsoft.DotNet.ApiCompat.Task
 From Version 9.0.100-alpha.1.23524.12 -> To Version 9.0.100-alpha.1.23525.7

* Update dependencies from https://github.com/dotnet/arcade build 20231025.2

Microsoft.DotNet.Arcade.Sdk , Microsoft.DotNet.Build.Tasks.Archives , Microsoft.DotNet.Build.Tasks.Feed , Microsoft.DotNet.Build.Tasks.Installers , Microsoft.DotNet.Build.Tasks.Packaging , Microsoft.DotNet.Build.Tasks.TargetFramework , Microsoft.DotNet.Build.Tasks.Templating , Microsoft.DotNet.Build.Tasks.Workloads , Microsoft.DotNet.CodeAnalysis , Microsoft.DotNet.GenAPI , Microsoft.DotNet.GenFacades , Microsoft.DotNet.Helix.Sdk , Microsoft.DotNet.PackageTesting , Microsoft.DotNet.RemoteExecutor , Microsoft.DotNet.SharedFramework.Sdk , Microsoft.DotNet.VersionTools.Tasks , Microsoft.DotNet.XUnitAssert , Microsoft.DotNet.XUnitConsoleRunner , Microsoft.DotNet.XUnitExtensions
 From Version 9.0.0-beta.23518.2 -> To Version 9.0.0-beta.23525.2

Dependency coherency updates

Microsoft.DotNet.XliffTasks
 From Version 1.0.0-beta.23516.1 -> To Version 1.0.0-beta.23523.1 (parent: Microsoft.DotNet.Helix.Sdk

* Update dependencies from https://github.com/dotnet/xharness build 20231026.1

Microsoft.DotNet.XHarness.CLI , Microsoft.DotNet.XHarness.TestRunners.Common , Microsoft.DotNet.XHarness.TestRunners.Xunit
 From Version 8.0.0-prerelease.23523.1 -> To Version 9.0.0-prerelease.23526.1

* Update dependencies from https://github.com/dotnet/sdk build 20231025.29

Microsoft.DotNet.ApiCompat.Task
 From Version 9.0.100-alpha.1.23524.12 -> To Version 9.0.100-alpha.1.23525.29

* Try inheriting from LongLivedMarshalByRefObject

* Resolve XUnit ambiguity

* Just disable the warning

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>
Co-authored-by: Andy Gocke <andy@commentout.net>

kunalspathak committed:
Revert "[arm64] Addressing modes for SIMD (#67490)" (#93981)

This reverts commit fa294c0494adbcb55f22411c9e8b0ba11a88af4b.

pavelsavara committed:
managed pool max threads

pavelsavara commented:
/azp run runtime-wasm

pavelsavara committed:
Revert "[browser][MT] enable all MT library tests on the runtime CI pipeline (#91536)"

pavelsavara commented:
/azp run runtime-wasm

buyaa-n committed:
Abstract out LocalBuilder, emit Locals with new ILGenerator (#93809)

* Abstract LocalBuilder, emit LocalBuilder in ILGenerator

* Apply suggestions from code review

Co-authored-by: Aaron Robinson <arobins@microsoft.com>

* Remove public LocalBuilder.Method, and apply other feedback

* Avoid invalid cast exception.

Co-authored-by: Jan Kotas <jkotas@microsoft.com>

---------

Co-authored-by: Aaron Robinson <arobins@microsoft.com>
Co-authored-by: Jan Kotas <jkotas@microsoft.com>

MichalPetryka committed:
Interlocked intrinsic improvements (#93821)

* Interlocked intrinsic improvements

* Format

EgorBo committed:
Fold "X >= 0 && X < NN" to "X u< NN" (#93834)

* Fold "X >= 0 && X < NN" to "X u< NN"

* Relax to GTF_SIDE_EFFECT

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/arcade dotnet/sdk (#94083)

* Update dependencies from https://github.com/dotnet/arcade build 20231027.1

Microsoft.DotNet.Arcade.Sdk , Microsoft.DotNet.Build.Tasks.Archives , Microsoft.DotNet.Build.Tasks.Feed , Microsoft.DotNet.Build.Tasks.Installers , Microsoft.DotNet.Build.Tasks.Packaging , Microsoft.DotNet.Build.Tasks.TargetFramework , Microsoft.DotNet.Build.Tasks.Templating , Microsoft.DotNet.Build.Tasks.Workloads , Microsoft.DotNet.CodeAnalysis , Microsoft.DotNet.GenAPI , Microsoft.DotNet.GenFacades , Microsoft.DotNet.Helix.Sdk , Microsoft.DotNet.PackageTesting , Microsoft.DotNet.RemoteExecutor , Microsoft.DotNet.SharedFramework.Sdk , Microsoft.DotNet.VersionTools.Tasks , Microsoft.DotNet.XUnitAssert , Microsoft.DotNet.XUnitConsoleRunner , Microsoft.DotNet.XUnitExtensions
 From Version 9.0.0-beta.23525.2 -> To Version 9.0.0-beta.23527.1

Dependency coherency updates

Microsoft.DotNet.XliffTasks
 From Version 1.0.0-beta.23523.1 -> To Version 1.0.0-beta.23525.5 (parent: Microsoft.DotNet.Helix.Sdk

* Update dependencies from https://github.com/dotnet/sdk build 20231027.4

Microsoft.DotNet.ApiCompat.Task
 From Version 9.0.100-alpha.1.23525.29 -> To Version 9.0.100-alpha.1.23527.4

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

vitek-karas committed:
ILLink: Allow interface methods to not have newslot (#93662)

Fixes https://github.com/dotnet/runtime/issues/93008

This comes from F#, where interface method defined in F# doesn't have `newslot` flag in its metadata. Trimmer gets confused and doesn't recognize such method as a "base" method for its implementation in the type which implements the interface. As a result, the implementation method is removed from the program because there doesn't seem to be a use for it anywhere.

The fix is to basically ignore the `newslot` flag for interface instance methods. The existing behavior must be kept for static interface methods though - as a redefinition of a method with the `new` C# keyword will produce an interface method definition without the `newslot` flag as well.

The change adds a new test and modifies the AOT/analyzer infra to run the test as well.

jkoritzinsky committed:
Implement the ComVariant struct, the ComVariantMarshaller type, and generator integration (#93635)

radical committed:
[wasm] Error out for AOT with no il trimming (#94064)

Without trimming AOT builds would be very slow, and unusable.

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/installer (#94084)

Microsoft.Dotnet.Sdk.Internal
 From Version 9.0.100-alpha.1.23525.7 -> To Version 9.0.100-alpha.1.23527.1

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

radekdoulik committed:
[wasm] make it easy to exclude browser-bench measurements (#94099)

Add options to the simple server, including the URL suffix

Add exclusions URL option to the browser-bench

This way we can exclude selected measurements. This can be
used for MT measurements, where we are hitting few crashes.

mrsharm committed:
Added the LOHThreshold public key to the GC Config. (#94068)

* Added the LOHThreshold public key

* Cleaned up formatting

ivdiazsa committed:
Add Separate Mode to XUnitLogChecker Where Only Dumps are Analyzed (#94094)

* Overhauled the XUnitLogChecker to be able to work without a log for
the libraries tests.

* Adjusted the Helix script in helixpublishwitharcade.proj to use the
new flagged command-line arguments, when calling the XUnitLogChecker.

t-mustafin committed:
[RISC-V] Disable FastTailCall in case of split arg (#93655)

wfurt committed:
remove extra assert from WinHttp (#93984)

radical committed:
[wasm] Update Wasm.Build.Tests to build with tfm=net9.0 (#93693)

* workload-testing.targets: Make dotnet script paths available outside

.. targets. This is useful when `dotnet-none` is already installed and
thus the installation target does not run.

* WorkloadManifest.targets: Disable workaround to use net9.0 artifacts

.. with `tfm=net8.0` . But don't disable this for Blazor projects which
are still using `tfm=net8.0`.

* [wasm] Improve parsing runtime pack versions from environment variables

* [wasm] Allow WBTOverridePack targets use to be set per test

This allows having it enabled only for the blazor projects right now,
which are still on tfm=net8.0 .

* [wasm] WBT: Update tests to run with net9.0

* [wasm] WBT: Remove --experiment-wasm-simd for v8

* [wasm] WBT: differentiate version used for tasks from the tfm

* [wasm] Fix wasmbrowser template so it correctly updates the tfm in csproj

* [wasm] WBT: Add runtime pack version for 9 in wasi

* [wasm] WBT: Share nuget config files between wasm and wasi

* [wasm] WBT: Fix blazor tests

* [wasm] WBT: Don't test 8.0 templates yet. this will need more changes

* [wasm] Don't default ForceNet8Current=false yet

Instead set this only for WBT.

* perf: add dummy change to trigger runtime-wasm-perf

agocke committed:
Update trim/aot analyzer to run on generated code (#94069)

lewing committed:
Use the native cross compilers on win-arm64 (#92443)

dotnet-maestro[bot] committed:
[main] Update dependencies from dnceng/internal/dotnet-optimization (#94030)

* Update dependencies from https://dev.azure.com/dnceng/internal/_git/dotnet-optimization build 20231025.5

optimization.linux-arm64.MIBC.Runtime , optimization.linux-x64.MIBC.Runtime , optimization.windows_nt-arm64.MIBC.Runtime , optimization.windows_nt-x64.MIBC.Runtime , optimization.windows_nt-x86.MIBC.Runtime , optimization.PGO.CoreCLR
 From Version 1.0.0-prerelease.23521.3 -> To Version 1.0.0-prerelease.23525.5

* Update dependencies from https://dev.azure.com/dnceng/internal/_git/dotnet-optimization build 20231025.5

optimization.linux-arm64.MIBC.Runtime , optimization.linux-x64.MIBC.Runtime , optimization.windows_nt-arm64.MIBC.Runtime , optimization.windows_nt-x64.MIBC.Runtime , optimization.windows_nt-x86.MIBC.Runtime , optimization.PGO.CoreCLR
 From Version 1.0.0-prerelease.23521.3 -> To Version 1.0.0-prerelease.23525.5

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

lewing committed:
Build the net8 workload msis (#94125)

maraf committed:
[browser][Wasm.Build.Tests] Use trimmer friendly JSON serialization (#94034)

* Use trimmer friendly JSON serialization

* CI: trigger only WBT on testasset changes

* CI: don't trigger non-wbt wasm jobs on testassets changes

---------

Co-authored-by: Ankit Jain <radical@gmail.com>

jkoritzinsky committed:
Remove the JIT/CheckProjects tool (#94121)

Fixes #76087

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/arcade (#94111)

* Update dependencies from https://github.com/dotnet/arcade build 20231027.3

Microsoft.DotNet.Arcade.Sdk , Microsoft.DotNet.Build.Tasks.Archives , Microsoft.DotNet.Build.Tasks.Feed , Microsoft.DotNet.Build.Tasks.Installers , Microsoft.DotNet.Build.Tasks.Packaging , Microsoft.DotNet.Build.Tasks.TargetFramework , Microsoft.DotNet.Build.Tasks.Templating , Microsoft.DotNet.Build.Tasks.Workloads , Microsoft.DotNet.CodeAnalysis , Microsoft.DotNet.GenAPI , Microsoft.DotNet.GenFacades , Microsoft.DotNet.Helix.Sdk , Microsoft.DotNet.PackageTesting , Microsoft.DotNet.RemoteExecutor , Microsoft.DotNet.SharedFramework.Sdk , Microsoft.DotNet.VersionTools.Tasks , Microsoft.DotNet.XUnitAssert , Microsoft.DotNet.XUnitConsoleRunner , Microsoft.DotNet.XUnitExtensions
 From Version 9.0.0-beta.23527.1 -> To Version 9.0.0-beta.23527.3

* React to xliff-tasks -> arcade repo consolidation

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>
Co-authored-by: Viktor Hofer <viktor.hofer@microsoft.com>

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/emsdk dotnet/sdk dotnet/arcade (#94136)

* Update dependencies from https://github.com/dotnet/arcade build 20231027.4

Microsoft.DotNet.Arcade.Sdk , Microsoft.DotNet.Build.Tasks.Archives , Microsoft.DotNet.Build.Tasks.Feed , Microsoft.DotNet.Build.Tasks.Installers , Microsoft.DotNet.Build.Tasks.Packaging , Microsoft.DotNet.Build.Tasks.TargetFramework , Microsoft.DotNet.Build.Tasks.Templating , Microsoft.DotNet.Build.Tasks.Workloads , Microsoft.DotNet.CodeAnalysis , Microsoft.DotNet.GenAPI , Microsoft.DotNet.GenFacades , Microsoft.DotNet.Helix.Sdk , Microsoft.DotNet.PackageTesting , Microsoft.DotNet.RemoteExecutor , Microsoft.DotNet.SharedFramework.Sdk , Microsoft.DotNet.VersionTools.Tasks , Microsoft.DotNet.XliffTasks , Microsoft.DotNet.XUnitAssert , Microsoft.DotNet.XUnitConsoleRunner , Microsoft.DotNet.XUnitExtensions
 From Version 9.0.0-beta.23527.3 -> To Version 9.0.0-beta.23527.4

* Update dependencies from https://github.com/dotnet/emsdk build 20231028.1

Microsoft.NET.Workload.Emscripten.Current.Manifest-9.0.100.Transport
 From Version 9.0.0-alpha.1.23523.3 -> To Version 9.0.0-alpha.1.23528.1

* Update dependencies from https://github.com/dotnet/sdk build 20231028.1

Microsoft.DotNet.ApiCompat.Task
 From Version 9.0.100-alpha.1.23527.4 -> To Version 9.0.100-alpha.1.23528.1

* Update dependencies from https://github.com/dotnet/arcade build 20231028.1

Microsoft.DotNet.Arcade.Sdk , Microsoft.DotNet.Build.Tasks.Archives , Microsoft.DotNet.Build.Tasks.Feed , Microsoft.DotNet.Build.Tasks.Installers , Microsoft.DotNet.Build.Tasks.Packaging , Microsoft.DotNet.Build.Tasks.TargetFramework , Microsoft.DotNet.Build.Tasks.Templating , Microsoft.DotNet.Build.Tasks.Workloads , Microsoft.DotNet.CodeAnalysis , Microsoft.DotNet.GenAPI , Microsoft.DotNet.GenFacades , Microsoft.DotNet.Helix.Sdk , Microsoft.DotNet.PackageTesting , Microsoft.DotNet.RemoteExecutor , Microsoft.DotNet.SharedFramework.Sdk , Microsoft.DotNet.VersionTools.Tasks , Microsoft.DotNet.XliffTasks , Microsoft.DotNet.XUnitAssert , Microsoft.DotNet.XUnitConsoleRunner , Microsoft.DotNet.XUnitExtensions
 From Version 9.0.0-beta.23527.3 -> To Version 9.0.0-beta.23528.1

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

liveans committed:
Disabling two socket tests (#93916)

* Disable two socket tests

* Revert close to dispose

* Disable tests against one issue

trylek committed:
Convert all tests under baseservices to the merged test infrastructure (#91560)

* Remove unused args parameter from the Main method in 349379.cs

* Convert stackoverflowtester Main to individual test entrypoints

* Don't complain about Exe type for test components

Some tests have exe components - if these aren't marked with
the CLRTestKind BuildAndRun, we shouldn't complain about them.

Thanks

Tomas

* Make dynamicmethodliveness and ParallelCrash merge-friendly

* Adjust the Tier1StackTrace test to be tolerant to merged wrappers

* Convert baseservices/exceptions to merged mode

* Remove constant return value 100 from the test test448035

* Make UnsafeAccessorTests owner class public

* Make methods in RuntimeConfiguration/TestConfig public

* Make TieredCompilation/BasicTest public

* Remove unused exit code of runmoduleconstructor

* Remove unused exit code of RuntimeHelperTests

* Fix visibility in multidimarray/enum test

* Fix visibility in TestCallingConventions test

* Fix visibility in CriticalFinalizer test

* Simplify RuntimeConfiguration/TestConfig

* Clean up TieredCompilation tests

* Convert istypeequivalent to use ConditionalFact clauses

* Fix visibility in RuntimeHelpersTests

* Add CoreCLRTestLibrary as a dependency of istypeequivalent

* Fix merged behavior of test448035

* Fix entrypoint in 305155

* Modify TestConfig to use a separate TestConfigTester app

* Additional fixes to TestConfig / TestConfigTester

* Mechanically merge all remaining tests under baseservices

* Fix BasicTestWithMcj, address initial Mark's PR feedback

* Remove superfluous OutputType=Library annotations per Marks' PR feedback

* Fix the baseservices/exceptions/unhandled test

* Fix stackoverflow3 and unhandled exception tests

* Remove unnecessary check from Directory.Build.targets

* Fix stackoverflowtester per Mark's PR feedback

TIHan committed:
Enable IlasmRoundTrip tests for merged tests (#93368)

* Initial work

* Tweak

* Tweak

* Do not generate _ilasmroundtrip.py for tests that have a generated run script

* Reduce imports

* Force fail to see where CI fails

* Revert forcing failure. Run roundtrip on build.

* Trying to fix script

* Backslash tweak

* Do not roundtrip the same assembly

* Fixing

* Remove import glob

* Added is_managed_assembly

* remove print

* Fixed paths

* Support bash. Ignore certain tests for arm. Fix poison test.

* Update CLRTest.Jit.targets

* Feedback. Added AssemblyChecker.

* Fix paths

* Update src/tests/Common/Directory.Build.targets

Co-authored-by: Bruce Forstall <brucefo@microsoft.com>

* Feedback

* Feedback

* Update AssemblyChecker.csproj

* Update Program.cs

* Trying to fix calling python on helix.

* Remove old roundtrip script calls. Added --is-exe option for AssemblyChecker.

* Tweak option:

* Remove check

* Remove imports

* Fix build

* Fix syntax errors. Fixed Popen arguments

* Fixed Popen arguments

* Fixing debug check

* Fixing tests

* Update ILVerificationTests.csproj

* Fixing tests

* Feedback

* Feedback

* Update CLRTest.Jit.targets

* Added help usage flag for AssemblyChecker

* Feedback on assembly-checker

---------

Co-authored-by: Bruce Forstall <brucefo@microsoft.com>

pedrobsaila committed:
Improve ArraySegment debugging (#90488)

gbalykov committed:
Fix test returning 100 unconditionally (#94146)

Xunit error XUW1002: Tests should not unconditionally return 100. Convert to a void return.

ilonatommy commented on src/libraries/Microsoft.Extensions.HostFactoryResolver/tests/HostFactoryResolverTests.cs:
```diff
@@ -279,7 +284,7 @@ public void ApplicationNameSetFromArgument()
         [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]
         [DynamicDependency(DynamicallyAccessedMemberTypes.All, typeof(NoSpecialEntryPointPattern.Program))]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/91538", typeof(PlatformDetection), nameof(PlatformDetection.IsWasmThreadingSupported))]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/73420", typeof(PlatformDetection), nameof(PlatformDetection.IsNativeAot))]
```
This is a closed issue, is it a typo or should we reopen it?

pavelsavara replied:
bad merge, sorry

ilonatommy commented on src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.cs:
```diff
@@ -17,7 +17,9 @@ internal sealed partial class PortableThreadPool
 
-#if TARGET_64BIT
+#if TARGET_BROWSER
+        private const short DefaultMaxWorkerThreadCount = 16;
```
not 32?

pavelsavara replied:
there is 
- pthread pool (emscripten level)
- managed thread pool (portion of pthread pool )
- threads allocated by user code (will use the rest of the available pthreads, but even that could be not enough)

ilonatommy reviewed (commented):
I checked and all removed issues are relevant

akoeplinger committed:
Delete Interop.PrintDlg file, it was moved to dotnet/winforms (#94171)

pavelsavara commented on eng/pipelines/extra-platforms/runtime-extra-platforms-wasm.yml:
```diff
@@ -112,14 +112,14 @@ jobs:
-       # NOTE - Since threading is experimental, we don't want to block mainline work
-      shouldContinueOnError: true
+      # NOTE - Since threading is experimental, we don't want to block mainline work
+      # TODO put back shouldContinueOnError: true
```
we should put this back before merge

Rob-Hague committed:
Use SequenceEqual in BigInteger.Equals (#91416)

eiriktsarpalis committed:
Implement PriorityQueue.Remove (#93994)

* Implement PriorityQueue.Remove

* Update src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

* Update src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

Co-authored-by: Dan Moseley <danmose@microsoft.com>

* Update src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

* Update src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

* Update src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

Co-authored-by: Stephen Toub <stoub@microsoft.com>

* Address feedback.

* Address feedback

* Add a Dijkstra smoke test.

* Alias distance type

---------

Co-authored-by: Dan Moseley <danmose@microsoft.com>
Co-authored-by: Stephen Toub <stoub@microsoft.com>

wfurt committed:
enable two HTTP POST tests (#94115)

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/arcade dotnet/emsdk dotnet/sdk dotnet/runtime dotnet/source-build-reference-packages (#94147)

* Update dependencies from https://github.com/dotnet/arcade build 20231028.2

Microsoft.DotNet.Arcade.Sdk , Microsoft.DotNet.Build.Tasks.Archives , Microsoft.DotNet.Build.Tasks.Feed , Microsoft.DotNet.Build.Tasks.Installers , Microsoft.DotNet.Build.Tasks.Packaging , Microsoft.DotNet.Build.Tasks.TargetFramework , Microsoft.DotNet.Build.Tasks.Templating , Microsoft.DotNet.Build.Tasks.Workloads , Microsoft.DotNet.CodeAnalysis , Microsoft.DotNet.GenAPI , Microsoft.DotNet.GenFacades , Microsoft.DotNet.Helix.Sdk , Microsoft.DotNet.PackageTesting , Microsoft.DotNet.RemoteExecutor , Microsoft.DotNet.SharedFramework.Sdk , Microsoft.DotNet.VersionTools.Tasks , Microsoft.DotNet.XliffTasks , Microsoft.DotNet.XUnitAssert , Microsoft.DotNet.XUnitConsoleRunner , Microsoft.DotNet.XUnitExtensions
 From Version 9.0.0-beta.23528.1 -> To Version 9.0.0-beta.23528.2

* Update dependencies from https://github.com/dotnet/emsdk build 20231028.4

Microsoft.NET.Workload.Emscripten.Current.Manifest-9.0.100.Transport
 From Version 9.0.0-alpha.1.23528.1 -> To Version 9.0.0-alpha.1.23528.4

* Update dependencies from https://github.com/dotnet/sdk build 20231028.2

Microsoft.DotNet.ApiCompat.Task
 From Version 9.0.100-alpha.1.23528.1 -> To Version 9.0.100-alpha.1.23528.2

* Update dependencies from https://github.com/dotnet/arcade build 20231028.2

Microsoft.DotNet.Arcade.Sdk , Microsoft.DotNet.Build.Tasks.Archives , Microsoft.DotNet.Build.Tasks.Feed , Microsoft.DotNet.Build.Tasks.Installers , Microsoft.DotNet.Build.Tasks.Packaging , Microsoft.DotNet.Build.Tasks.TargetFramework , Microsoft.DotNet.Build.Tasks.Templating , Microsoft.DotNet.Build.Tasks.Workloads , Microsoft.DotNet.CodeAnalysis , Microsoft.DotNet.GenAPI , Microsoft.DotNet.GenFacades , Microsoft.DotNet.Helix.Sdk , Microsoft.DotNet.PackageTesting , Microsoft.DotNet.RemoteExecutor , Microsoft.DotNet.SharedFramework.Sdk , Microsoft.DotNet.VersionTools.Tasks , Microsoft.DotNet.XliffTasks , Microsoft.DotNet.XUnitAssert , Microsoft.DotNet.XUnitConsoleRunner , Microsoft.DotNet.XUnitExtensions
 From Version 9.0.0-beta.23528.1 -> To Version 9.0.0-beta.23528.2

* Update dependencies from https://github.com/dotnet/emsdk build 20231028.4

Microsoft.NET.Workload.Emscripten.Current.Manifest-9.0.100.Transport
 From Version 9.0.0-alpha.1.23528.1 -> To Version 9.0.0-alpha.1.23528.4

* Update dependencies from https://github.com/dotnet/sdk build 20231030.1

Microsoft.DotNet.ApiCompat.Task
 From Version 9.0.100-alpha.1.23528.1 -> To Version 9.0.100-alpha.1.23530.1

* Update dependencies from https://github.com/dotnet/runtime build 20231029.4

Microsoft.DotNet.ILCompiler , Microsoft.NET.Sdk.IL , Microsoft.NETCore.App.Runtime.win-x64 , Microsoft.NETCore.ILAsm , runtime.native.System.IO.Ports , System.Text.Json
 From Version 9.0.0-alpha.1.23523.1 -> To Version 9.0.0-alpha.1.23529.4

* Update dependencies from https://github.com/dotnet/source-build-reference-packages build 20231027.1

Microsoft.SourceBuild.Intermediate.source-build-reference-packages
 From Version 9.0.0-alpha.1.23519.2 -> To Version 9.0.0-alpha.1.23527.1

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

AndyAyersMS committed:
JIT: update tail call IR validity checks (#94130)

Remove `fgCheckStmtAfterTailCall` as it did less thorough and less
correct checking.

Contributes to #93246.

elinor-fung committed:
Remove BundleLegacy test targeting netcoreapp3.0/3.1 (#94118)

dotnet-maestro[bot] committed:
[main] Update dependencies from dotnet/installer (#94137)

* Update dependencies from https://github.com/dotnet/installer build 20231027.23

Microsoft.Dotnet.Sdk.Internal
 From Version 9.0.100-alpha.1.23527.1 -> To Version 9.0.100-alpha.1.23527.23

* Update dependencies from https://github.com/dotnet/installer build 20231028.2

Microsoft.Dotnet.Sdk.Internal
 From Version 9.0.100-alpha.1.23527.1 -> To Version 9.0.100-alpha.1.23528.2

* Update dependencies from https://github.com/dotnet/installer build 20231029.1

Microsoft.Dotnet.Sdk.Internal
 From Version 9.0.100-alpha.1.23527.1 -> To Version 9.0.100-alpha.1.23529.1

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

buyaa-n committed:
Emit/Call member references in new ILGenerator (#94116)

* Emit/Call member references in IL

* Add more tests, add double iterations for prepopulating tokens

* Emit constructor reference, add more tests

* Apply feedbacks

kouvel committed:
Expose a `Lock` type in preview mode (#87672)

* Expose a `Lock` type in preview mode

- For now, the `Lock` type requires preview features to be enabled
- Ported CoreCLR's AwareLock implementation to C# with a bit of refactoring, folded in a couple of ideas from NativeAOT's previous `Lock` implementation, and fixed a couple of issues.
- Added an adaptive spin strategy to reduce CPU time from spin-waiting when spin-waits are not effective
- This implementation replaces NativeAOT's `Lock` implementation. The performance of acquiring a lock under contention is improved in NativeAOT.

API review: https://github.com/dotnet/runtime/issues/34812

TIHan committed:
Arm64: Implement LoadVector64x*AndUnzip and LoadVector128x*AndUnzip APIs (#94128)

* Initial LoadVector64/128AndUnzip APIs

* Added hwintrinsics

* Added generating tests

* Fix semi

* Whitespace

* Whitespace

* Whitespace

* Whitespace

* Whitespace

* Whitespace

* Fix test generation

* fix missing cases in gentree

* Minor formatting of the test generator

* Fix comments

* Fixing up tests

* Use the right register variants

brantburnett committed:
Use Span to fill List<T> in more ToList scenarios (#86796)

* Use Span to fill List<T> in more ToList scenarios

* Optimize Append/Prepend changes

* Update src/libraries/System.Linq/src/System/Linq/Lookup.SpeedOpt.cs

Co-authored-by: Stephen Toub <stoub@microsoft.com>

* Update src/libraries/System.Linq/src/System/Linq/Lookup.SpeedOpt.cs

Co-authored-by: Stephen Toub <stoub@microsoft.com>

* Address feedback

* Remove ToSingleItemList, seems like an overoptimization

---------

Co-authored-by: Stephen Toub <stoub@microsoft.com>

MichalStrehovsky committed:
Avoid handing out duplicate `CORINFO_MODULE_STRUCT_*` handles (#94082)

RyuJIT depends on never seeing two different `CORINFO_MODULE_STRUCT` for the same thing.

Fixes #93843.

TIHan committed:
ARM64: Fix Vector64x2/3/4 API comments. Minor cleanup in AdvSimd. (#94188)

* Fix Vector64x2/3/4 API comments. Minor cleanup in AdvSimd.

* 4h to 4H

ilonatommy committed:
Merge branch 'browser_mt_pool_size' of https://github.com/pavelsavara/runtime into browser_mt_pool_size

ilonatommy committed:
Revert merge bug.
